ts/defs.bzl

```diff
@@ -309,14 +313,14 @@ def ts_project(
     typings_out_dir = declaration_dir if declaration_dir else out_dir
     tsbuildinfo_path = ts_build_info_file if ts_build_info_file else name + ".tsbuildinfo"
 
-    tsc_typings_outs = _lib.calculate_typings_outs(srcs, typings_out_dir, root_dir, declaration, composite, allow_js)
-    tsc_typing_maps_outs = _lib.calculate_typing_maps_outs(srcs, typings_out_dir, root_dir, declaration_map, allow_js)
+    tsc_typings_outs = _lib.calculate_typings_outs(srcs, typings_out_dir, root_dir, declaration, composite, allow_js, no_emit)
+    tsc_typing_maps_outs = _lib.calculate_typing_maps_outs(srcs, typings_out_dir, root_dir, declaration_map, allow_js, no_emit)
 
     tsc_js_outs = []
     tsc_map_outs = []
-    if not transpiler or transpiler == "tsc":
-        tsc_js_outs = _lib.calculate_js_outs(srcs, out_dir, root_dir, allow_js, resolve_json_module, preserve_jsx, emit_declaration_only)
-        tsc_map_outs = _lib.calculate_map_outs(srcs, out_dir, root_dir, source_map, preserve_jsx, emit_declaration_only)
+    if no_emit or not transpiler or transpiler == "tsc":
```

Comment:
This logic change feels suspicious to me. What's the use case for going into this branch with noEmit? Is it correct to allow no_emit with a custom transpiler, or should we just `fail` in that case?

Reply:
This was just copied from the other PR and I agree it's odd.

I think we can change this and then also revert the changes to `calculate_*_outs`.

Reply:
`no_emit` + `transpiler` just means typecheck-only - a single rule with a validation action and no outputs (other then the validation action stdout file). I think that's valid?

Reply:
UPDATE: I think we should keep this as-is for now in this PR. Going into this `if` branch to only use tsc is easiest for now as it completely avoids changing the `else` branch for now - that `else` will change significantly when `isolated_declarations` is added so I'd like to wait until then.

Reply:
okay great, thanks for investigating it